### DISCUSSION TITLE
About vector graphics

### DISCUSSION THREAD

**Discussion by iacore** [david-vanderson/dvui#42] [General]
Is it possible to use plutovg-like API to draw custom vector graphics in a fixed-sized region?

For example, 

```zig
var canvas = dvui.canvas(.{.w=100, .h=100});
// canvas.ctx should be, for example, plutovg
canvas.ctx.moveTo(10, 10);
canvas.ctx.lineTo(20, 20);
canvas.draw();
```

I have a planned feature in my REPL application.

What I need to do for minimal PoC:

- reserve a fixed-sized area
- paint the area white
- draw a black line in that area. the visual should be clipped inside the area.

**Comment by iacore** [comment]
![image](https://github.com/david-vanderson/dvui/assets/74560659/92a81da5-55f9-486e-827b-37c72b08ecd3)
I've made it!
so the usage is that every two-element array like `[0 1]` has a vector display.

This is for debugging, of course. I'll add a tab widget like the one in gtoolkit later.

***

Here's a screenshot of gtoolkit. The view tabs ("Repositories", "JSON String") provide different views for the same data.

![gtoolkit screenshot](https://github.com/david-vanderson/dvui/assets/74560659/2f133f2d-a69f-4335-b35c-9a60c0e06d52)

**Comment by david-vanderson** [comment]
That's awesome!!
